source-stripe-native/source_stripe_native/resources.py

```diff
@@ -2,140 +2,102 @@
 from typing import AsyncGenerator, Awaitable, Iterable
 from logging import Logger
 import functools
+import requests
+
+from requests.auth import HTTPBasicAuth
 
 from estuary_cdk.flow import CaptureBinding
 from estuary_cdk.capture import Task
 from estuary_cdk.capture.common import Resource, LogCursor, PageCursor, open_binding, ResourceConfig, ResourceState
 from estuary_cdk.http import HTTPSession, HTTPMixin, TokenSource
 
-from .api import (fetch_incremental,
-                  fetch_backfill,
-                  fetch_incremental_substreams,
-                  fetch_backfill_substreams,
-                  fetch_incremental_no_events,
-                  fetch_backfill_usage_records,
-                  fetch_incremental_usage_records,
-                  )
+from .api import (
+    API,
+    fetch_incremental,
+    fetch_backfill,
+    fetch_incremental_substreams,
+    fetch_backfill_substreams,
+    fetch_incremental_no_events,
+    fetch_backfill_usage_records,
+    fetch_incremental_usage_records,
+)
 
 from .models import (
-    Accounts,
-    Authorizations,
-    Bank_Accounts,
-    BalanceTransactions,
-    Cards,
-    CardHolders,
-    Coupons,
-    Customers,
-    CustomerBalanceTransaction,
-    Charges,
-    CheckoutSessions,
-    CheckoutSessionsLine,
-    CreditNotes,
-    CreditNotesLines,
-    Disputes,
-    EarlyFraudWarning,
-    Persons,
-    Invoices,
-    InvoiceItems,
-    InvoiceLineItems,
-    ExternalAccountCards,
-    ExternalBankAccount,
-    ApplicationFees,
-    ApplicationFeesRefunds,
-    Bank_Accounts,
-    PaymentMethods,
-    PaymentIntent,
-    Payouts,
-    Plans,
-    Products,
-    PromotionCode,
-    Refunds,
-    Reviews,
-    SetupAttempts,
-    SetupIntents,
-    Subscriptions,
-    SubscriptionsSchedule,
-    SubscriptionItems,
-    TopUps,
-    Transactions,
-    Transfers,
-    TransferReversals,
-    UsageRecords,
-    Files,
-    FilesLink,
     EndpointConfig,
-    EventResult,
-    )
+    STREAMS,
+    REGIONAL_STREAMS,
+)
+
+
+def check_accessibility(
+    access_token: str, url: str,
+) -> bool:
+    """
+    Returns a boolean representing whether the provided stream is accessible (True if accessible, False if inaccessible).
+
+    The Stripe API response has a 200 or 404 status code if the stream is accessible.
+
+    404 status codes are received when checking a child stream that can be inaccessible when its parent stream is accessible.
+    """
+    code = requests.get(url=url, auth=HTTPBasicAuth(access_token, "")).status_code
+    return code == 200 or code == 404
```

Comment:
I think we need to explicitly check for a specific code that means the stream is _not_ accessible, like a `400` code. Otherwise intermittent errors may falsely claim that the stream is not accessible, and the binding could get disabled from an autodiscovery.

I'd also suggest using the `http` object we pass around everywhere for this - that way its retying mechanisms can be used.

Reply:
I'll update to check for the status code returned when a stream isn't accessible (IIRC the Stripe API returns a `403`, but I'll confirm).

I initially didn't use the `http` object we pass around because the [error it raises](https://github.com/estuary/connectors/blob/164f6284e2ec7742f053328bd3206828049ce475/estuary-cdk/estuary_cdk/http.py#L279-L281) doesn't provide the status code as a separate property, but rather lists it the error message. I didn't like the idea of keying off of a message that could change at a later time. 

What do you think about me updating those lines in our CDK to raise a custom error like the one below instead? It'd provide the status code as a dedicated property instead of stuffed inside the error message.
```python
class HTTPError(RuntimeError):
    def __init__(self, message: str, code: int):
        super().__init__(message)
        self.code = code
```

Reply:
Yeah that seems great!